feat: add shortcut completion support

doodleEsc opened the pull request:
- Update README to reflect the addition of a new shortcut completion triggered by `#`
- Add new configuration options and examples for customizing shortcut icons and names in the README
- Introduce a new `default_shortcut_get_items` function to handle shortcut completions
- Add a new `default_shortcut_get_label` function to format shortcut labels
- Integrate shortcut functionality into the main configuration, including triggers, item retrieval, labeling, kind naming, insertion text, and documentation

NanoBoom committed:
feat: add shortcut completion support

- Update README to reflect the addition of a new shortcut completion triggered by `#`
- Add new configuration options and examples for customizing shortcut icons and names in the README
- Introduce a new `default_shortcut_get_items` function to handle shortcut completions
- Add a new `default_shortcut_get_label` function to format shortcut labels
- Integrate shortcut functionality into the main configuration, including triggers, item retrieval, labeling, kind naming, insertion text, and documentation

Signed-off-by: doodleEsc <cinuor@gmail.com>

Kaiser-Yang commented:
I've add a .stylua.toml to master branch, can you just rebase your branch and then use the stylua to format the files you updated with the new .stylua.toml?

NanoBoom committed:
refactor: refactor code style for consistency and brevity

- Refactor code to use single-line function definitions
- Update string literals from double quotes to single quotes for consistency
- Shorten if-statements to be single-line
- Reformat `should_show_items` function for improved readability

Signed-off-by: doodleEsc <cinuor@gmail.com>

Kaiser-Yang commented:
Thank you for your contribution.
